spec/std/json/serializable_spec.cr

```diff
@@ -471,6 +471,25 @@ class JSONSomething
   property value : JSONSomething?
 end
 
+module JsonDiscriminatorBug
+  abstract class Base
+    include JSON::Serializable
+
+    use_json_discriminator("type", {"a" => A, "b" => B, "c" => C})
+  end
+
+  class A < Base
+  end
```

Comment:
`A` is not necessary for a minimal reproduction

Reply:
A need because it leaf node, without it would infinite recursion.